packages/components/src/components/Log/Log.js

```diff
@@ -28,17 +33,92 @@ const itemSize = 15; // This should be kept in sync with the line-height in SCSS
 const defaultHeight = itemSize * 100 + itemSize / 2;
 
 export class LogContainer extends Component {
-  state = { loading: true };
+  constructor(props) {
+    super(props);
+    this.state = { loading: true };
+    this.logRef = React.createRef();
+    this.textRef = React.createRef();
+  }
 
   componentDidMount() {
     this.loadLog();
+    if (this.props.enableLogAutoScroll) {
+      this.wasRunningAfterMounting();
+      window.addEventListener('scroll', this.handleLogScroll, true);
+      this.handleLogScroll();
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      this.props.enableLogAutoScroll &&
+      (prevState.logs?.length !== this.state.logs?.length ||
+        prevProps.isLogsMaximized !== this.props.isLogsMaximized)
+    ) {
+      if (this.shouldAutoScroll()) {
+        this.scrollToBottomLog();
+        return;
+      }
+      this.handleLogScroll();
+    }
   }
 
   componentWillUnmount() {
+    window.removeEventListener('scroll', this.handleLogScroll, true);
     clearInterval(this.timer);
     this.cancelled = true;
   }
 
+  handleLogScroll = () => {
+    const isLogBottomUnseen = this.isLogBottomUnseen();
+
+    if (isLogBottomUnseen !== this.state.isLogBottomUnseen) {
+      this.setState({
+        isLogBottomUnseen
+      });
+    }
+  };
+
+  shouldAutoScroll = () => {
+    return (
+      this.props.enableLogAutoScroll &&
+      !this.state.isLogBottomUnseen &&
+      this.wasRunningAfterMounting() &&
+      this.isLogBottomUnseen()
+    );
+  };
+
+  isLogBottomUnseen = () => {
+    return (
+      isElementEndBelowViewBottom(this.logRef?.current) ||
+      hasElementPositiveVerticalScrollBottom(
+        this.textRef?.current?.firstElementChild
+      )
+    );
+  };
+
+  scrollToBottomLog = () => {
+    const longTextElement = this.textRef?.current?.firstElementChild;
+    if (longTextElement) {
+      longTextElement.scrollTop =
+        longTextElement.scrollHeight - longTextElement.clientHeight;
+    }
+    const rootElement = document.documentElement;
+    rootElement.scrollTop = rootElement.scrollHeight - rootElement.clientHeight;
+  };
+
+  wasRunningAfterMounting = () => {
+    if (this.alreadyWasRunningAfterMounting) {
+      return true;
+    }
+    const { reason, status } = getStepStatusReason(this.props.stepStatus);
+    if (isRunning(reason, status)) {
+      this.alreadyWasRunningAfterMounting = true;
```

Comment:
this should be set in state

Reply:
As discussed, I added a comment in the amended commit explaining that this is a class variable because it does not require a subsequent re-rendering